docs/guides/modals.md

```diff
@@ -0,0 +1,409 @@
+* Only the inputs in `"type": "input"` blocks will be included in `view.state.values`
+* Each input/selection in non-`"input"` typed blocks such as `"section"`, `"actions"` is sent as a `block_actions` request
+* You use `callback_id` to identify a modal, a pair of `block_id` and `action_id` to identify each input in `view.state.values`
+* You can use `view.private_metadata` to hold the internal state and/or `block_actions` results on the modal
+* You respond to `view_submission` requests with `response_action` to determine the next state of a modal
+* **views.update/push** APIs are supposed to be used when receiving `block_actions` requests in modals, NOT for `view_submission` requests
+
+## Bolt Examples
+
+**NOTE**: If you're a beginner to using Bolt for Slack App development, consult [The Basics of Bolt]({{ site.url | append: site.baseurl }}/guides/Bolt), first.
+
+Let's start with opening a modal. Let's say, we're going to open the following modal.
+
+```javascript
+{
+  "callback_id": "meeting-arrangement",
+  "type": "modal",
+  "notify_on_close": true,
+  "title": { "type": "plain_text", "text": "Meeting Arrangement", "emoji": true },
+  "submit": { "type": "plain_text", "text": "Submit", "emoji": true },
+  "close": { "type": "plain_text", "text": "Cancel", "emoji": true },
+  "blocks": [
+    {
+      "block_id": "agenda-block",
+      "type": "input",
+      "element": { "action_id": "agenda-action", "type": "plain_text_input", "multiline": true },
+      "label": { "type": "plain_text", "text": "Detailed Agenda", "emoji": true }
+    }
+  ]
+}
+```
+
+**slack-api-client** offers a smooth DSL for building blocks and views. The following code generates **View** objects in a type-safe way.
+
+```java
+import com.slack.api.model.view.View;
+import static com.slack.api.model.view.Views.*;
+import static com.slack.api.model.block.Blocks.*;
+import static com.slack.api.model.block.composition.BlockCompositions.*;
+import static com.slack.api.model.block.element.BlockElements.*;
+
+View buildView() {
+  return view(view -> view
+    .callbackId("meeting-arrangement")
+    .type("modal")
+    .notifyOnClose(true)
+    .title(viewTitle(title -> title.type("plain_text").text("Meeting Arrangement").emoji(true)))
+    .submit(viewSubmit(submit -> submit.type("plain_text").text("Submit").emoji(true)))
+    .close(viewClose(close -> close.type("plain_text").text("Cancel").emoji(true)))
+    .blocks(asBlocks(
+      input(input -> input
+        .blockId("agenda-block")
+        .element(plainTextInput(pti -> pti.actionId("agenda-action").multiline(true)))
+        .label(plainText(pt -> pt.text("Detailed Agenda").emoji(true)))
+      )
+    ))
+  );
+}
+```
+
+If you need to carry some information to the modal, setting `private_metadata` is a good way for it. The `private_metadata` is a single string data (3,000 characters at maximum). So, if you have multiple values, you need to serialize them into a string in a format.
```

Comment:
suggestion: "is a single string data (3,000 characters at maximum)" -> "is a single string with a maximum of 3000 characters"